src/metal.cs

```diff
@@ -2892,8 +3173,158 @@ interface MTLTileRenderPipelineDescriptor : NSCopying {
+		nuint MaxTotalThreadsPerThreadgroup { get; set; }
+
 		[Export ("reset")]
 		void Reset ();
 	}
+
+	interface IMTLEvent {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[Protocol]
+	interface MTLEvent {
+		[Abstract]
+		[NullAllowed, Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DesignatedDefaultCtor]
+	interface MTLSharedEventListener {
+		[Export ("initWithDispatchQueue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (DispatchQueue dispatchQueue);
+
+		[Export ("dispatchQueue")]
+		DispatchQueue DispatchQueue { get; }
+	}
+
+	delegate void MTLSharedEventNotificationBlock (IMTLSharedEvent arg0, ulong arg1);
+
+	interface IMTLSharedEvent {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[Protocol]
+	interface MTLSharedEvent : MTLEvent {
+		[Abstract]
+		[Export ("notifyListener:atValue:block:")]
+		void NotifyListener (MTLSharedEventListener listener, ulong atValue, MTLSharedEventNotificationBlock block);
+
+		[Abstract]
+		[Export ("newSharedEventHandle")]
+		MTLSharedEventHandle CreateSharedEventHandle ();
+
+		[Abstract]
+		[Export ("signaledValue")]
+		ulong SignaledValue { get; set; }
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MTLSharedEventHandle : NSSecureCoding {
+		[NullAllowed, Export ("label")]
+		string Label { get; }
+	}
+
+	interface IMTLIndirectRenderCommand {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[Protocol]
+	interface MTLIndirectRenderCommand {
+		[NoiOS, NoTV, Mac (10,14, onlyOn64: true)]
+		[Abstract]
+		[Export ("setRenderPipelineState:")]
+		void SetRenderPipelineState (IMTLRenderPipelineState pipelineState);
+
+		[Abstract]
+		[Export ("setVertexBuffer:offset:atIndex:")]
+		void SetVertexBuffer (IMTLBuffer buffer, nuint offset, nuint index);
+
+		[Abstract]
+		[Export ("setFragmentBuffer:offset:atIndex:")]
+		void SetFragmentBuffer (IMTLBuffer buffer, nuint offset, nuint index);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV]
+		[Export ("drawPatches:patchStart:patchCount:patchIndexBuffer:patchIndexBufferOffset:instanceCount:baseInstance:tessellationFactorBuffer:tessellationFactorBufferOffset:tessellationFactorBufferInstanceStride:")]
+		void DrawPatches (nuint numberOfPatchControlPoints, nuint patchStart, nuint patchCount, [NullAllowed] IMTLBuffer patchIndexBuffer, nuint patchIndexBufferOffset, nuint instanceCount, nuint baseInstance, IMTLBuffer buffer, nuint offset, nuint instanceStride);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV]
+		[Export ("drawIndexedPatches:patchStart:patchCount:patchIndexBuffer:patchIndexBufferOffset:controlPointIndexBuffer:controlPointIndexBufferOffset:instanceCount:baseInstance:tessellationFactorBuffer:tessellationFactorBufferOffset:tessellationFactorBufferInstanceStride:")]
```

Comment:
Yay Apple for coming up with a selector that doesn't fit on my iMac Pro screen 🙄